storage/watchable_store.go

```diff
@@ -356,3 +360,25 @@ type watching struct {
 	// The chan might be shared with other watchings.
 	ch chan<- storagepb.Event
 }
+
+// safelySynced safely puts watching with key k into watchableStore's synced.
+func safelySynced(s *watchableStore, k string, wa *watching) error {
+	if v, ok := s.synced[k]; ok {
+		if wa != nil {
+			if _, ok := v[wa]; ok {
+				return fmt.Errorf("put the same watch twice: %+v", wa)
+			} else {
+				v[wa] = struct{}{}
+			}
+		}
+		if len(s.synced[k]) == 0 {
```

Comment:
well... I do not think this should be here. this should be where you REMOVE a watching from the map.